slider/replay.py

```diff
@@ -134,19 +137,24 @@ def _consume_actions(buffer):
     decompressed_data = lzma.decompress(compressed_data)
 
     out = []
+    abs_offset = 0
     for raw_action in decompressed_data.split(b','):
         if not raw_action:
             continue
-        offset, x, y, raw_action_mask = raw_action.split(b'|')
+        raw_offset, x, y, raw_action_mask = raw_action.split(b'|')
         action_mask = ActionBitMask.unpack(int(raw_action_mask))
+        rel_offset = int(raw_offset)
+        abs_offset += rel_offset
         out.append(Action(
-            datetime.timedelta(milliseconds=int(offset)),
+            datetime.timedelta(milliseconds=rel_offset),
```

Comment:
I've never been a fan of using timedeltas at all because I've always felt osu times really are in milliseconds but I understand why you do that, (and if I was using this for anything other than what I am using it for I probably would prefer them) so feel free to change it. One thing though: does numpy's version have the .total_seconds method? I'm using that to convert to ms and I'd like to keep using it (or something analogous to it)